### DISCUSSION TITLE
Support for Tezos?

### DISCUSSION THREAD

**Discussion by LordDarkHelmet** [hdwallet-io/python-hdwallet#21] [Ideas]
I am learning about working with Tezos (XTZ) HD wallets. Your project has broad support for many currencies, but Tezos does not seem to be one on your list. Is this something easy to add? Looking through the code I can see that adding coins happens in  `hdwallet/cryptocurrencies.py` 

It looks like you would just add something like this:

```Python
class TezosMainnet(Cryptocurrency):

    NAME = "Tezos"
    SYMBOL = "XTZ"
    NETWORK = "mainnet"
    SOURCE_CODE = "https://gitlab.com/tezos/tezos"
    COIN_TYPE = CoinType({
        "INDEX": 1729,
        "HARDENED": True
    })

    SCRIPT_ADDRESS = 0x??
    PUBLIC_KEY_ADDRESS = 0??
    SEGWIT_ADDRESS = SegwitAddress({
        "HRP": None,
        "VERSION": 0x00
    })

    EXTENDED_PRIVATE_KEY = ExtendedPrivateKey({
        "P2PKH": 0x????????,
        "P2SH": 0x????????,
        "P2WPKH": None,
        "P2WPKH_IN_P2SH": None,
        "P2WSH": None,
        "P2WSH_IN_P2SH": None
    })
    EXTENDED_PUBLIC_KEY = ExtendedPublicKey({
        "P2PKH": 0x????????,
        "P2SH": 0x????????,
        "P2WPKH": None,
        "P2WPKH_IN_P2SH": None,
        "P2WSH": None,
        "P2WSH_IN_P2SH": None
    })

    MASSAGE_PREFIX = "\x18Tezos Signed Message:\n"
    DEFAULT_PATH = f"m/44'/{str(COIN_TYPE)}/0'/0/0"
    WIF_SECRET_KEY = 0x??
```

Clearly from the above I do not have all of the details as I have not used HD wallets with Tezos before. Is there documentation on how to derive/find/add the missing variables? 

http://tezos-wiki.jp/wiki/index.php?title=List_of_address_prefixes (looks like BTC's values)
https://gitlab.com/tezos/tezos/blob/master/src/lib_crypto/base58.ml

It looks like there is no standard WIF format in Tezos.

**Comment by meherett** [comment]
Okay, I will implement it in the next version but, you can also implement it. After you finish just create a pull request for this.
